src/background.js

```diff
@@ -884,77 +603,94 @@ function updateCount(details){
     return {};
   }
 
-  if(!Utils.isPrivacyBadgerEnabled(getHostForTab(details.tabId))){
+  if(!Utils.isPrivacyBadgerEnabled(pb.webrequest.getHostForTab(details.tabId))){
     return;
   }
 
   var tabId = details.tabId;
-  if (!tabData[tabId]) {
+  if (!pb.tabData[tabId]) {
     return;
   }
-  if(tabData[tabId].bgTab === true){
+  if(pb.tabData[tabId].bgTab === true){
     // prerendered tab, Chrome will throw error for setBadge functions, don't call
     return;
-  }else if(tabData[tabId].bgTab === false){
-    updateBadge(tabId);
-  }else{
+  } else {
     chrome.tabs.get(tabId, function(tab){
       if (chrome.runtime.lastError){
-        tabData[tabId].bgTab = true;
-      }else{
-        tabData[tabId].bgTab = false;
+        pb.tabData[tabId].bgTab = true;
+      } else {
+        pb.tabData[tabId].bgTab = false;
         updateBadge(tabId);
       }
     });
   }
 }
-chrome.webRequest.onBeforeRequest.addListener(updateCount, {urls: ["http://*/*", "https://*/*"]}, []);
-
 
 /**
-* Populate tabs object with currently open tabs when extension is updated or installed. 
-*/
-function updateTabList(){
-  console.log('update tabs!');
-  // Initialize the tabData/frames object if it is falsey
-  tabData = tabData || {};
-  chrome.tabs.query({currentWindow: true, status: 'complete'}, function(tabs){
-    for(var i = 0; i < tabs.length; i++){
-      var tab = tabs[i];
-      tabData[tab.id] = {
-        frames: {
-          0: {
-            parent: -1,
-            url: tab.url
-          }
-        },
-        domains: {}
-      };
+ * Check if a specific frame is whitelisted
+ * TODO: used in popup-blocker.js inspect if necessary
+ *
+ * @param {Integer} tabId The id of the tab
+ * @param {Integer} frameId The id of the frame
+ * @param {String} type Content type to be checked
+ * @returns {boolean} true if whitelisted
+ */
+function isFrameWhitelisted(tabId, frameId, type) {
+  var parent = frameId;
+  var parentData = webrequest.getFrameData(tabId, parent);
+  while (parentData)
+  {
+    var frameData = parentData;
+
+    parent = frameData.parent;
+    parentData = webrequest.getFrameData(tabId, parent);
+
+    var frameUrl = frameData.url;
+    var parentUrl = (parentData ? parentData.url : frameUrl);
+    if ("keyException" in frameData || isWhitelisted(frameUrl)){
```

Comment:
Where else would we keep it? This is checking if a domain is unblocked essentially, not if privacy badger is disabled.